internal/controller/storageclient_controller.go

```diff
@@ -369,43 +393,43 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 
 	r.storageClient.Status.InMaintenanceMode = storageClientResponse.MaintenanceMode
 
-	for _, kubeResource := range storageClientResponse.KubeResources {
-		var err error
-		typeMeta := &metav1.TypeMeta{}
-		if err := json.Unmarshal(kubeResource, typeMeta); err != nil {
-			return reconcile.Result{}, fmt.Errorf("failed to unmarshal the type for the Object: %w", err)
-		}
-
-		// Create the received resources, if necessary.
-		switch typeMeta.GroupVersionKind() {
-		case quotav1.SchemeGroupVersion.WithKind("ClusterResourceQuota"):
-			err = r.reconcileResource(&quotav1.ClusterResourceQuota{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("CephConnection"):
-			err = r.reconcileResource(&csiopv1a1.CephConnection{}, kubeResource)
-		case csiopv1a1.GroupVersion.WithKind("ClientProfileMapping"):
-			err = r.reconcileResource(&csiopv1a1.ClientProfileMapping{}, kubeResource)
-		case corev1.SchemeGroupVersion.WithKind("Secret"):
-			err = r.reconcileResource(&corev1.Secret{}, kubeResource)
-		case nbv1.SchemeGroupVersion.WithKind("Noobaa"):
-			err = r.reconcileResource(&nbv1.NooBaa{}, kubeResource)
-		case storagev1.SchemeGroupVersion.WithKind("StorageClass"):
-			err = r.reconcileResource(&storagev1.StorageClass{}, kubeResource)
-		case snapapi.SchemeGroupVersion.WithKind("VolumeSnapshotClass"):
-			err = r.reconcileResource(&snapapi.VolumeSnapshotClass{}, kubeResource)
-		case groupsnapapi.SchemeGroupVersion.WithKind("VolumeGroupSnapshotClass"):
-			if val, _ := r.crdsBeingWatched.Load(VolumeGroupSnapshotClassCrdName); !val.(bool) {
+	requestsPerGvk := map[string][]reconcileRequest{}
+	for _, kubeObj := range storageClientResponse.KubeObjects {
+		if kubeObj != nil {
```

Comment:
Let's loos one indentation
```suggestion
		if kubeObj == nil {
			continue
		}
		
```